pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java

```diff
@@ -154,6 +159,8 @@ public static void createHelixClusterIfNeeded(String helixClusterName, String zk
         .buildEmptyIdealStateForBrokerResource(admin, helixClusterName, enableBatchMessageMode);
     admin.setResourceIdealState(helixClusterName, CommonConstants.Helix.BROKER_RESOURCE_INSTANCE, idealState);
     initPropertyStorePath(helixClusterName, zkPath);
+
+    createLeadControllerResourceIfNeeded(helixClusterName, admin, enableBatchMessageMode, leadControllerResourceConfig);
```

Comment:
can we do this outside this method and have the caller call createLeadControllerResourceIfNeeded?

Reply:
The reason why I put it inside `createHelixClusterIfNeeded` is that `createHelixClusterIfNeeded` method can only focus on setting up cluster steps, whereas `startHelixControllerInStandadloneMode` method can only focus on initializing controller.